File: LeetCode/Basic/169/main.py
from typing import List
from collections import Counter


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        m = Counter()
        for num in nums:
            m[num] += 1

        ans = 0
        size = len(nums)//2
        for k, v in m.items():
            if size < v:
                ans = k

        return ans

File: LeetCode/Basic/169/main.go
package main

func majorityElement(nums []int) int {
	var ans int
	m := make(map[int]int)
	for _, num := range nums {
		m[num]++
	}

	border := len(nums) / 2
	for k, cnt := range m {
		if cnt > border {
			ans = k
		}
	}
	return ans
}
